source/Messages.cpp

```diff
@@ -15,15 +15,21 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 
 #include "Messages.h"
 
+#include "Color.h"
+#include "GameData.h"
+
 #include <mutex>
 
 using namespace std;
 
 namespace {
+	const int MAX_LOG = 10000;
+
 	mutex incomingMutex;
 
 	vector<pair<string, Messages::Importance>> incoming;
-	vector<Messages::Entry> list;
+	vector<Messages::Entry> recent;
+	vector<pair<string, Messages::Importance>> log;
```

Comment:
clang doesn't seem to like this name (because it's ambiguous with `std:log`) - see CI failure. Guess you'll have to rename it